File: smoke/python/smoke.py
"""Live smoke: one POST /v1/responses via PyPI-layout Python package (`pypi/src/zerogpu`)."""
from __future__ import annotations

import os
import sys
from pathlib import Path

from dotenv import load_dotenv

ROOT = Path(__file__).resolve().parents[2]
# Use the standalone package tree (import `zerogpu`), not raw `sdks/python` (flat Fern
# layout shadows the stdlib `types` module when placed first on sys.path).
PY_SRC = ROOT / "pypi" / "src"
sys.path.insert(0, str(PY_SRC))

for env_path in (
    Path(__file__).resolve().parent / ".env",
    Path(__file__).resolve().parents[3] / "Benchmark" / ".env",
):
    load_dotenv(env_path)


def main() -> None:
    from zerogpu import ZerogpuApi

    api_key = (os.environ.get("ZEROGPU_API_KEY") or "").strip()
    project_id = (os.environ.get("ZEROGPU_PROJECT_ID") or "").strip()
    model = (os.environ.get("ZEROGPU_MODEL") or "").strip()
    if not api_key or not project_id:
        print("Set ZEROGPU_API_KEY and ZEROGPU_PROJECT_ID", file=sys.stderr)
        sys.exit(1)
    if not model:
        print("Set ZEROGPU_MODEL", file=sys.stderr)
        sys.exit(1)

    prompt = (os.environ.get("ZEROGPU_INPUT_TEXT") or "").strip() or (
        "In one short sentence, what is a habit tracker?"
    )

    client = ZerogpuApi(project_id=project_id, api_key=api_key)
    res = client.responses.create_response(
        model=model,
        input=prompt,
        text={"format": {"type": "text"}},
    )
    print("id:", res.id)
    print("model:", res.model)
    print("usage:", res.usage)


if __name__ == "__main__":
    main()

File: smoke/ruby/smoke.rb
# frozen_string_literal: true

# Live smoke: one POST /v1/responses via Fern-generated Ruby SDK.
# Requires Ruby >= 3.3 (see sdks/ruby/zerogpu.gemspec).

require "bundler/setup"
require "zerogpu"

def load_dotenv_file(path)
  return unless File.file?(path)

  File.foreach(path) do |line|
    line = line.strip
    next if line.empty? || line.start_with?("#")

    k, _, v = line.partition("=")
    k = k.strip
    v = v.strip.sub(/\A['"]/, "").sub(/['"]\z/, "")
    ENV[k] = v unless ENV[k] && !ENV[k].empty?
  end
end

root = File.expand_path("../..", __dir__)
load_dotenv_file(File.join(__dir__, ".env"))
load_dotenv_file(File.expand_path("../Benchmark/.env", root))

api_key = ENV.fetch("ZEROGPU_API_KEY", "").strip
project_id = ENV.fetch("ZEROGPU_PROJECT_ID", "").strip
model = ENV.fetch("ZEROGPU_MODEL", "").strip
if api_key.empty? || project_id.empty?
  warn "Set ZEROGPU_API_KEY and ZEROGPU_PROJECT_ID"
  exit 1
end
if model.empty?
  warn "Set ZEROGPU_MODEL"
  exit 1
end

prompt = ENV.fetch("ZEROGPU_INPUT_TEXT", "").strip
prompt = "In one short sentence, what is a habit tracker?" if prompt.empty?

client = Zerogpu::Client.new(api_key: api_key)

res = client.responses.create_response(
  request_options: { additional_headers: { "x-project-id" => project_id } },
  model: model,
  input: [{ role: Zerogpu::Types::InputMessageRole::USER, content: prompt }],
  text: { format: { type: "text" } }
)

puts "id: #{res.id}"
puts "model: #{res.model}"
puts "usage: #{res.usage.inspect}"
